### PR TITLE
Fix incorrect process name cutting on macos

### LINKED ISSUE #49: Fails to show process name sometimes
The #30 introduced a bug that results in `name` sometimes being incorrect.

At least on mac, the issue is that the output of `ps` can get truncated when asking for all the info in a single `ps` run.

For example `ps uwwxo pid,pid,ppid,uid,%cpu,%mem,comm,args` returns this line for a certain process:

> rafal  3897   0.0  0.0 408700720    736 s003  S    Mon11AM   0:05.50 /opt/homebrew/Ce  3897  3897  3884   501   0.0  0.0 /opt/homebrew/Ce /opt/homebrew/Cellar/powerlevel10k/1.15.0/gitstatus/usrbin/gitstatusd-darwin-arm64 -G v1.5.1 -s -1 -u -1 -d -1 -c -1 -m -1 -v FATAL -t 16

while the "non-optimized" query `ps wwxo pid,comm` returns:

>  3897 /opt/homebrew/Cellar/powerlevel10k/1.15.0/gitstatus/usrbin/gitstatusd-darwin-arm64

So you can see that the "combined" query is truncating the `comm` thus `ps-list` is failing to obtain the correct process name from that.

@stroncium

### PR DESCRIPTION
**Pull request by jopemachine**
I found `ps-list` returns incorrect process name while working on https://github.com/sindresorhus/fkill-cli/pull/83

I think it is broken in ps-list@7.0.0 (https://github.com/sindresorhus/ps-list/blob/v7.0.0/index.js#L125).

This PR will fix https://github.com/sindresorhus/fkill-cli/issues/78.

Only tested in macOS yet.

## After

afte

---

Fixes #49

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -81,6 +81,8 @@ const psFields = 'pid,ppid,uid,%cpu,%mem,comm,args';
 
 const psOutputRegex = /^[ \t]*(?<pid>\d+)[ \t]+(?<ppid>\d+)[ \t]+(?<uid>\d+)[ \t]+(?<cpu>\d+\.\d+)[ \t]+(?<memory>\d+\.\d+)[ \t]+/;
 
+const flagsRegex = /--.*/;
```
Comment (sindresorhus) [inline index.js:84]:
```suggestion
const flagsRegex = / --.*/;
```

File: index.js [hunk 2]
```diff
@@ -126,10 +128,9 @@ const nonWindowsSingleCall = async (options = {}) => {
 		throw new Error(ERROR_MESSAGE_PARSING_FAILED);
 	}
 
-	const commLength = argsPosition - commPosition;
 	for (const [index, line] of lines.entries()) {
-		processes[index].name = line.slice(commPosition, commPosition + commLength).trim();
 		processes[index].cmd = line.slice(argsPosition).trim();
+		processes[index].name = path.basename(processes[index].cmd).replaceAll(flagsRegex, '');
```
Comment (sindresorhus) [inline index.js:133]: No reason to use `replaceAll`

File: index.js [hunk 3]
```diff
@@ -126,10 +128,9 @@ const nonWindowsSingleCall = async (options = {}) => {
 		throw new Error(ERROR_MESSAGE_PARSING_FAILED);
 	}
 
-	const commLength = argsPosition - commPosition;
 	for (const [index, line] of lines.entries()) {
```
Comment (sindresorhus) [inline index.js:131]:
```suggestion
	for (const [index, line] of lines.entries()) {
	  const process = processes[index];
```

**Comment by jopemachine** [comment]
I'm so sorry, my patch was totally wrong.

Using replaceAll with incorrect regexp (should be with g flag) caused TypeError and the error calls `nonWindowsMultipleCalls`, that's why it passed the tests.

The cutting bug is not caused by fkill's logic, it is by `ps` itself.

For example, I found `ps wwxo comm,args` command prints below.

```
...
/Applications/Vi /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/jopemachine/.vscode/extensions/samverschueren.linter-xo-3.14.1/dist/server.js --node-ipc --clientProcessId=17051
...
```
`comm` here is `/Applications/Vi`, which is already cut off.

I found this posting, https://unix.stackexchange.com/questions/401780/ps-cuts-command-how-to-get-full

This posting said to swap the argument order, but it didn't work since `comm` and `args` both are too long.

I think the sole solution here is calling `ps wwxo args` separately with `ps wwxo pid,ppid,uid,%cpu,%mem,comm`.

It increases the runtime cost, but still better than calling `nonWindowsMultipleCalls`, and it enables to return correct names.

**Comment by sindresorhus** [comment]
Yes, that is correct.

I should never have merged https://github.com/sindresorhus/ps-list/commit/285f434b1114823a5cc1cec710ab8253ca36cebc

**Comment by sindresorhus** [comment]
Have you tested these changes with `fkill-cli`?

**Comment by jopemachine** [comment]
> Have you tested these changes with `fkill-cli`?

Yes, and I also have tested below simple test code.

```js
import psList from './index.js';
console.log(await psList({all: true}));
```

**Pull request merged**